### PR TITLE
Allow errors to show technical message on demand

### PR DESCRIPTION
**Pull request by jleach**
# Summary of Changes

Allow `BifoldError` to show a technical message to help debugging as desired by the user.

# Related Issues

n/a

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [x] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [x] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [x] Updated documentation as needed for changed code and new or modified features;
- [x] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

**Comment by jleach** [comment]
@PatStLouis  There are two lines added to `App/localization/fr/index.ts`, able to suggest the proper translation?

File: core/App/localization/en/index.ts [hunk 1]
```diff
@@ -27,7 +27,8 @@ const translation = {
     "Okay": "Okay",
     "GoBack": "Go Back",
     "GetStarted": "Get Started",
-    "Dismiss": "Dismiss"
+    "Dismiss": "Dismiss",
+    "ShowDetails": "Show Details",
```
Comment (PatStLouis) [inline core/App/localization/en/index.ts:31]: To include in the fr localization file: "Afficher les détails"

File: core/App/localization/en/index.ts [hunk 2]
```diff
@@ -36,6 +37,7 @@ const translation = {
   "Error": {
     "Unknown": "Unknown Error",
     "Problem": "A problem has occurred",
+    "ErrorCode": "Error code",
```
Comment (PatStLouis) [inline core/App/localization/en/index.ts:40]: To include in the fr localization file: "Code d'erreur"

**Comment by PatStLouis** [comment]
Hey @jleach , the **fr** file wasn't included in the PR so I've commented the relevant lines in the **en** file with the suggestions. Let me know if this works for you.

File: core/App/localization/en/index.ts (hunk 1 above)
Comment (jleach) [reply core/App/localization/en/index.ts:31]: Added.

File: core/App/localization/en/index.ts (hunk 2 above)
Comment (jleach) [reply core/App/localization/en/index.ts:40]: Added

**Review by amanji — APPROVED** [review]

**Pull request merged**